### DISCUSSION TITLE
Freerdp server for wayland

### DISCUSSION THREAD

**Discussion by Sanjubisanal** [FreeRDP/FreeRDP#7212] [Q&A]
Can **freerdp-shadow-cli** be used as rdp server for wayland display server.

**Comment by bmiklautz** [comment]
Currently the shadow-cli supports only X11 but unfortunately not yet wayland. Not sure it is still available and if it helps but there was native support in Weston for access over RDP with FreeRDP.

**Reply by Sanjubisanal** [reply]
Yes weston has rdp-backend that uses freerdp for transport but that is specific to weston. But in my case I have custom wayland based display manager, so need generic rdp server for wayland. anyway thank you for your kind information.

**Reply by hardening** [reply]
A generic wayland backend for the shadow server is very unlikely to happen as it entirely depends on the used compositor (and in most cases it's just impossible to implement). If you handle the compositor yourself (custom display manager), then if you use libweston you may get the RDP backend "for free". You can also have a look at gnome remote desktop that is wayland native and use FreeRDP for its remoting.

**Reply by Sanjubisanal** [reply]
@hardening @bmiklautz  Thank you for your direction.

**Reply by bmiklautz** [reply]
> A generic wayland backend for the shadow server is very unlikely to happen as it entirely depends on the used compositor (and in most cases it's just impossible to implement). If you handle the compositor yourself (custom display manager), then if you use libweston you may get the RDP backend "for free". You can also have a look at gnome remote desktop that is wayland native and use FreeRDP for its remoting.

Am I mistaken or isn't there a common interface for screen shot/access/sharing over D-Bus/PipeWire or is this also composer specific?

**Reply by hardening** [reply]
AFAICT all these are compositor specific and most of the time involve implementing a specific wayland protocol (when possible). The easiest to implement is when the compositor does screen-shooting / recording itself.
